src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -406,14 +406,31 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
         );
     }
 
-    fn visit_operand(&mut self, op: &Operand<'tcx>, location: Location) {
-        self.super_operand(op, location);
-        if let Operand::Constant(c) = op {
-            if let Some(def_id) = c.check_static_ptr(self.tcx) {
-                self.check_static(def_id, self.span);
+    fn visit_constant(&mut self, constant: &Constant<'tcx>, location: Location) {
+        self.super_constant(constant, location);
+
+        if let ty::ConstKind::Unevaluated(def_id, _, promoted) = constant.literal.val {
+            assert!(promoted.is_none(), "Const-checking should run before promotion");
+
+            // If a cyclic data dependency exists within a const initializer, try to find
+            // it during const-checking. This is important because MIR optimizations could
+            // eliminate a cycle before const-eval runs. See #71078 for an example of this.
```

Comment:
#71078 isn't actually an example of this, as Ralf notes.

Reply:
Huh? It isn't? Or are you talking about a comment by Ralf from elsewhere but this PR?